File: ep1-bip39-python/src/check_mnemonic.py
from mnemonic import Mnemonic

good_mnemonic = "monkey frog radio rubber weekend doctor leave project stay junior dad catch"
bad_mnemonic = "frog radio rubber weekend doctor leave project stay junior dad catch monkey"
print("good mnemonic:", Mnemonic().check(good_mnemonic))
print("bad mnemonic:", Mnemonic().check(bad_mnemonic))

File: ep2-bip39-rust/src/bin/check_mnemonic.rs
use bip39::Mnemonic;

fn main() {
    check("monkey frog radio rubber weekend doctor leave project stay junior dad catch");
    check("frog radio rubber weekend doctor leave project stay junior dad catch monkey");
}

fn check(mnemonic: &str) {
    println!("mnemonic: {}", mnemonic);

    match Mnemonic::parse(mnemonic) {
        Ok(_) => {
            println!("valid")
        }
        Err(err) => {
            println!("invalid: {:?}", err)
        }
    }
}
